src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java

```diff
@@ -15,18 +15,16 @@
 
 import java.util.Optional;
 
-public class PlayerInteractListener implements Listener {
+public class    PlayerInteractListener implements Listener {
 
     @EventHandler(ignoreCancelled = true)
     public void onClick(PlayerInteractEvent event) {
         Player player = event.getPlayer();
         User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
         ItemStack itemInHand = player.getInventory().getItemInMainHand();
 
-        if (!(IridiumSkyblock.getInstance().getConfiguration().obsidianBucket
-                && event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
-                && event.getClickedBlock().getType().equals(Material.OBSIDIAN)
-                && itemInHand.getType().equals(Material.BUCKET))) {
+        if (!(IridiumSkyblock.getInstance().getConfiguration().obsidianBucket)
+            && !(IridiumSkyblock.getInstance().getConfiguration().endPortalPick)) {
             return;
         }
```

Comment:
This code is redundant since we do similar checks later on

Reply:
weirdly enough the code wouldn't function if I didn't add them, I'll see what I can do